Print monster death messages in the correct order

nstoddard opened the pull request:
Currently, the message when, for example, a bloat bursts is printed before the message saying what killed it. This reorders the messages in cases like this, so that they're printed in the correct order.

Fixes part (but not all) of #371.

tmewett commented:
I just rebased onto 'master' and added a commit. What do you think? I'm not sure the refreshSidebar did anything and it probably makes sense to call anyoneWantABite in killCreature?

nstoddard commented:
Your changes look okay to me (other than the mistake which I fixed a few minutes ago), and the result is cleaner than adding a new function for such a specific purpose, but it's harder to be confident that the new version doesn't break anything. I tested your changes with a few recordings from 1.11.1 (after rebasing a new branch back on `release`), and none of them desynced, so it's probably fine.

I don't actually remember why I added the `refreshSidebar` call; it seems to work fine without it. Maybe I added it just because it was called in `inflictDamage` after the monster was killed?
